gpytorch/lazy/kronecker_product_added_diag_lazy_tensor.py

```diff
@@ -57,6 +57,24 @@ def _logdet(self):
                 first_term = (const_times_evals.diag() + 1).log().sum(dim=-1)
                 return diag_term + first_term
 
+            else:
+               # we use the same matrix determinant identity: |K + D| = |D| |I + D^{-1}K|
+               # but have to symmetrize the second matrix because torch.eig may not be
+               # completely differentiable.
+                kron_times_diag_list = [
+                    tfull.matmul(tdiag.inverse()) for tfull, tdiag in zip(lt.lazy_tensors, dlt.lazy_tensors)
+                ]
+                
+                # We symmetrize the sub-components K_i D_i^{-1}
+                kron_times_diag_symm = KroneckerProductLazyTensor(
+                    *[k.matmul(k.transpose(-1, -2)) for k in kron_times_diag_list]
+                )
+                evals_square, _ = kron_times_diag_symm.symeig(eigenvectors=True)
+                evals_plus_i = DiagLazyTensor(evals_square.sqrt() + 1)
+                
+                diag_term = self.diag_tensor.diag().clamp(min=1e-7).log().sum(dim=-1)
+                return diag_term + evals_plus_i.sum(dim=-1)
```

Comment:
should we just call ._logdet()` on `evals_plus_i` here instead?

Reply:
yeah, we could do that. and diag_term.logdet() analogously.